upup/pkg/kutil/rollingupdate_cluster_test.go

```diff
@@ -0,0 +1,754 @@
+				Role: kopsapi.InstanceGroupRoleMaster,
+			},
+		},
+		Ready: []*CloudInstanceGroupInstance{
+			{
+				ASGInstance: &autoscaling.Instance{
+					InstanceId: aws.String("master-1a"),
+				},
+				Node: &v1.Node{},
+			},
+		},
+	}
+
+	groups["bastion-1"] = &CloudInstanceGroup{
+		ASGName: aws.StringValue(asgGroups.AutoScalingGroups[3].AutoScalingGroupName),
+		asg:     asgGroups.AutoScalingGroups[3],
+		InstanceGroup: &kopsapi.InstanceGroup{
+			ObjectMeta: api.ObjectMeta{
+				Name: "bastion-1",
+			},
+			Spec: kopsapi.InstanceGroupSpec{
+				Role: kopsapi.InstanceGroupRoleBastion,
+			},
+		},
+		Ready: []*CloudInstanceGroupInstance{
+			{
+				ASGInstance: &autoscaling.Instance{
+					InstanceId: aws.String("bastion-1a"),
+				},
+				Node: &v1.Node{},
+			},
+		},
+	}
+
+	err := c.RollingUpdate(groups, k8sClient)
+	if err != nil {
+		t.Errorf("Error on rolling update: %v", err)
+	}
+
+	asgGroups, _ = cloud.Autoscaling().DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
+		AutoScalingGroupNames: []*string{aws.String("node-1")},
+	})
+	for _, group := range asgGroups.AutoScalingGroups {
+		if len(group.Instances) != 2 {
+			t.Errorf("Expected 2 instances got: %v", len(group.Instances))
+		}
+	}
+
+	asgGroups, _ = cloud.Autoscaling().DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
+		AutoScalingGroupNames: []*string{aws.String("node-2")},
+	})
+	for _, group := range asgGroups.AutoScalingGroups {
+		if len(group.Instances) != 2 {
+			t.Errorf("Expected 2 instances got: %v", len(group.Instances))
+		}
+	}
+
+	asgGroups, _ = cloud.Autoscaling().DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
+		AutoScalingGroupNames: []*string{aws.String("master-1")},
+	})
+	for _, group := range asgGroups.AutoScalingGroups {
+		if len(group.Instances) != 1 {
+			t.Errorf("Expected 1 instances got: %v", len(group.Instances))
+		}
+	}
+
+	asgGroups, _ = cloud.Autoscaling().DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
+		AutoScalingGroupNames: []*string{aws.String("bastion-1")},
+	})
+	for _, group := range asgGroups.AutoScalingGroups {
+		if len(group.Instances) != 1 {
+			t.Errorf("Expected 1 instances got: %v", len(group.Instances))
+		}
+	}
+}
+
+func TestRollingUpdateNoneNeedUpdateWithForce(t *testing.T) {
+	k8sClient := fake.NewSimpleClientset()
+
+	mockcloud := awsup.BuildMockAWSCloud("us-east-1", "abc")
+	mockcloud.MockAutoscaling = &mockautoscaling.MockAutoscaling{}
+
+	c := &RollingUpdateCluster{
+		Cloud:           mockcloud,
+		MasterInterval:  1 * time.Millisecond,
+		NodeInterval:    1 * time.Millisecond,
+		BastionInterval: 1 * time.Millisecond,
+		Force:           true,
+	}
+	cloud := c.Cloud.(awsup.AWSCloud)
+	setUpCloud(c)
+
+	asgGroups, _ := cloud.Autoscaling().DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{})
+
+	groups := make(map[string]*CloudInstanceGroup)
+	groups["node-1"] = &CloudInstanceGroup{
```

Comment:
I don't think so since the `groups` var is what determines the outcome of the tests and should be hardcoded in each test case since it can vary. You'll notice in some tests the `NeedUpdate` fields are left blank.